bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemOffsetProfile.java

```diff
@@ -0,0 +1,137 @@
+            } catch (IllegalArgumentException e) {
+                logger.error("Cannot convert value '{}' of parameter '{}' into a valid offset of type QyantityType.",
+                        paramValue, OFFSET_PARAM);
+            }
+        } else {
+            logger.error("Parameter '{}' is not of type String", OFFSET_PARAM);
+        }
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return SystemProfiles.OFFSET;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+        logger.debug("State update from item: '{}'", state);
+        callback.handleCommand((Command) applyOffset(state, false));
+    }
+
+    @Override
+    public void onCommandFromItem(Command command) {
+        logger.debug("Command from item: '{}'", command);
+        callback.handleCommand((Command) applyOffset(command, false));
+    }
+
+    @Override
+    public void onCommandFromHandler(Command command) {
+        logger.debug("Command from handler: '{}'", command);
+        callback.sendCommand((Command) applyOffset(command, true));
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State state) {
+        logger.debug("State update from handler: '{}'", state);
+        callback.sendUpdate((State) applyOffset(state, true));
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private Type applyOffset(Type state, boolean towardsThing) {
+        QuantityType finalOffset = offset;
+        if (finalOffset == null) {
+            logger.error("Offset not configured correctly, returning undef state.");
+            return UnDefType.UNDEF;
+        }
+
+        if (!towardsThing) {
+            finalOffset = finalOffset.negate();
+        }
+
+        Type result = UnDefType.UNDEF;
+
+        if (state instanceof QuantityType) {
+            QuantityType qtState = (QuantityType) state;
+            try {
+                if (finalOffset.getUnit() == AbstractUnit.ONE) {
+                    // allow offsets without unit -> implicitly assume its the same as the one from the state
+                    // TODO: do we want this? User would have to look up the original unit on the channel in the
+                    // binding...
+                    finalOffset = new QuantityType<>(finalOffset.toBigDecimal(), qtState.getUnit());
+                }
+                result = qtState.add(finalOffset);
+            } catch (UnconvertibleException e) {
+                logger.error("Cannot apply offset '{}' to state '{}' because types do not match.", finalOffset,
+                        qtState);
+            }
+        } else if (state instanceof DecimalType && finalOffset.getUnit().equals(AbstractUnit.ONE)) {
+            DecimalType decState = (DecimalType) state;
+            result = new DecimalType(decState.doubleValue() + finalOffset.doubleValue());
+        } else {
+            logger.error("State '{}' of type '{}' is not applicable for this profile.", state, state.getClass());
```

Comment:
If I configure an offset of `-1 K` for a binding which doesn't yet support quantity types, I will also end up with this log entry. And this is pretty misleading for the user.

Reply:
Why is this misleading? A user wants to subtract `1 K` from a value on that channel, how should the framework know "how much" is should subtract in case the binding sends "23" but means "kg" ? Shall it subtract "1" and ignore the unit completely?

Reply:
Okay, let me give you an example:

```
... ERROR  State 23 of type DecimalType is not applicable for this profile.
```

How is this helpful? How is the user supposed to know how to handle this situation? Which profile? Which profile? And where does this state come from?